src/core/core.animator.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+
+const helpers = require('../helpers/index');
+
+function drawFPS(chart, count, date, lastDate) {
+	const fps = (1000 / (date - lastDate)) | 0;
+	const ctx = chart.ctx;
+	ctx.save();
+	ctx.clearRect(0, 0, 50, 24);
+	ctx.fillStyle = 'black';
+	ctx.textAlign = 'right';
+	if (count) {
+		ctx.fillText(count, 50, 8);
+		ctx.fillText(fps + ' fps', 50, 18);
+	}
+	ctx.restore();
+}
+
+class Animator {
+	constructor() {
+		this._request = null;
+		this._charts = new Map();
+		this._running = false;
+	}
+
+	/**
+	 * @private
+	 */
+	_notify(anims, type, args) {
+		const callbacks = anims._listeners[type] || [];
+		callbacks.forEach(fn => fn(args || []));
+	}
+
+	/**
+	 * @private
+	 */
+	_refresh() {
+		const me = this;
+
+		if (me._request) {
+			return;
+		}
+		me._running = true;
+
+		me._request = helpers.requestAnimFrame.call(window, function() {
+			me._update();
+			me._request = null;
+
+			if (me._running) {
+				me._refresh();
+			}
+		});
+	}
+
+	/**
+	 * @private
+	 */
+	_update() {
+		const date = Date.now();
+		const charts = this._charts;
+		let remaining = 0;
+
+		for (let [chart, anims] of charts) {
+			if (!anims.running || !anims.items.length) {
+				continue;
+			}
+			const items = anims.items;
+			let i = items.length - 1;
+			let draw = false;
+			let item;
+
+			for (; i >= 0; --i) {
+				item = items[i];
+
+				if (item._active) {
+					item.tick(date);
+					draw = true;
+				} else {
+					// Remove the item by replacing it with last item and removing the last
+					// A lot faster than splice.
+					items[i] = items[items.length - 1];
+					items.pop();
+				}
+			}
+
+			if (draw) {
+				chart.draw();
+				if (chart.options.animation.fps) {
```

Comment:
Yeah, I wanted to see this while testing and put it behind a option to hide it from fixture tests. This can be removed completely, its just a test / debug / demo thing.

Reply:
I'm fine to leave it if it's helpful for development. But we should probably either add to the dev documentation or go ahead and remove it